EIPS/eip-3584.md

```diff
@@ -0,0 +1,122 @@
+---
+eip: 3584
+title: Block Access List
+author: Gajinder Singh (@g11in), Piper Merriam (@pipermerriam)
+discussions-to: https://ethresear.ch/t/block-access-list-v0-1/9505
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-05-22
+requires: 2929, 2930
+---
+
+## Simple Summary
+A proposal to build a block's `access_list` and include its fingerprint `AccessListRoot` in the block header.
+
+## Abstract
+[EIP-2929](./eip-2929)/[EIP-2930](./eip-2930) centers around normalizing the (low) gas costs of data/storage accesses made by a transaction as well as providing for (and encouraging) a new transaction type format:
+```
+0x01 || rlp([chainId, nonce, gasPrice, gasLimit, to, value, data, access_list, yParity, senderR, senderS])
+```
+that makes upfront `access_list` declarations, where `access_list` is some `[[{20 bytes}, [{32 bytes}...]]...]` map of `AccessedAddress=> AccessedStorageKeys`.
+
+The first *accesses* of these upfront *declarations* are charged at discounted price (roughly ~`10%`) and first accesses outside this list are charged higher price. Reason is upfront access declaration provides for a way to *preload/optimize/batch* loading these locations while executing the transaction.
+This inadvertently leads to generation of transaction `access_list` that has all (first) accesses (declared or not) made by a transaction.
+This proposal is to collate these *transaction* `access_list`s for all the transactions in a **block** `access_list` document and include its *fingerprint* in the block header.
+
+## Motivation
+Motivation for collating the *transaction* `access_list`s for all the transactions in a **block**’s `access_list` is to have an *access index* of the block with following benefits:
+1. Block execution/validation optimizations/parallelization/cache warmup by enabling construction of *a partial order* for access and hence execution (hint: *chains* in this *poset* can be parallelized).
+2. Enabling partial inspection and fetching/serving of a block data/state by *light sync* or *fast sync* protocols concerned with a subset of addresses.
+3. Possible future extension of this list to serve as index for bundling, serving and fetching witness data for *stateless* protocols.
+
+## Specification
+A block `access_list` represents:
+```
+Set [ 
+	AccessedAddress, 
```

Comment:
Mixed tabs and spaces (looks like spaces everywhere else, but there is a tab leading this line).